tests/DCPS/UnitTests/Serializer.cpp

```diff
@@ -0,0 +1,298 @@
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_CDR_LE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::FINAL));
+  EXPECT_STREQ("CDR/XCDR1 Little Endian Plain", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_PL_CDR_BE_MUTABLE)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_PL_CDR_BE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::MUTABLE));
+  EXPECT_STREQ("CDR/XCDR1 Big Endian Parameter List", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_PL_CDR_LE_MUTABLE)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_PL_CDR_LE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::MUTABLE));
+  EXPECT_STREQ("CDR/XCDR1 Little Endian Parameter List", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_CDR2_BE_FINAL)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_CDR2_BE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::FINAL));
+  EXPECT_STREQ("XCDR2 Big Endian Plain", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_CDR2_LE_FINAL)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_CDR2_LE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::FINAL));
+  EXPECT_STREQ("XCDR2 Little Endian Plain", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_D_CDR2_BE_APPENDABLE)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_D_CDR2_BE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::APPENDABLE));
+  EXPECT_STREQ("XCDR2 Big Endian Delimited", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_D_CDR2_LE_APPENDABLE)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_D_CDR2_LE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::APPENDABLE));
+  EXPECT_STREQ("XCDR2 Little Endian Delimited", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_PL_CDR2_BE_MUTABLE)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_PL_CDR2_BE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::MUTABLE));
+  EXPECT_STREQ("XCDR2 Big Endian Parameter List", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_PL_CDR2_LE_MUTABLE)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_PL_CDR2_LE);
+  Encoding enc;
+  EXPECT_TRUE(eh.to_encoding(enc,Extensibility::MUTABLE));
+  EXPECT_STREQ("XCDR2 Little Endian Parameter List", eh.to_string().c_str());
+}
+
+TEST(serializer_test, EncapsulationHeader__to_encoding_XML)
+{
+  EncapsulationHeader eh;
+  eh.kind(EncapsulationHeader::Kind::KIND_XML);
+  Encoding enc;
+  EXPECT_FALSE(eh.to_encoding(enc,Extensibility::MUTABLE));
```

Comment:
I think this reports an warning message, so we might have to either suppress the scoreboard from reporting this specific warning or set the debug level to hide it.

Reply:
> I think this reports an warning message, so we might have to either suppress the scoreboard from reporting this specific warning or set the debug level to hide it.

It's an `ERROR` level. I wasn't aware the scoreboard would check the log levels. I'll look into suppressing it.

Reply:
Added `DCPS_debug_level` to suppress output.